marimo/_plugins/ui/_impl/table.py

```diff
@@ -440,18 +448,32 @@ def _get_banner_text(self) -> str:
         return ""
 
     def _convert_value(
-        self, value: Union[list[int], list[str]]
-    ) -> Union[list[JSONType], IntoDataFrame]:
-        indices = [int(v) for v in value]
-        if self._has_stable_row_id:
-            # Search across the original data
-            self._selected_manager = self._manager.select_rows(indices)
+        self, value: Union[list[int], list[str], list[dict[str, Any]]]
+    ) -> Union[list[JSONType], IntoDataFrame, list[Cell]]:
+        # To not break the initial selected rows value
+        # We transform a list of int to row coordinates.
+        if all(isinstance(v, int) for v in value):
+            value = [{"rowId": v} for v in value]
+
+        if self._selection in ["single-cell", "multi-cell"]:
+            coordinates = [
+                TableCoordinate(rowId=v["rowId"], columnName=v["columnName"])
+                for v in value
+                if isinstance(v, dict) and "rowId" in v and "columnName" in v
+            ]
+            self._has_any_selection = len(coordinates) > 0
+            return self._searched_manager.select_cells(coordinates)  # type: ignore
         else:
-            self._selected_manager = self._searched_manager.select_rows(
-                indices
-            )
-        self._has_any_selection = len(indices) > 0
-        return unwrap_narwhals_dataframe(self._selected_manager.data)  # type: ignore[no-any-return]
+            indices = [int(v["rowId"]) for v in value]
```

Comment:
we are doing some unnecessary transformation to dictionary and back. if its a list of ints. lets just keep it that way